graphql-dxm-provider/src/main/java/org/jahia/modules/graphql/provider/dxm/util/ZipUtils.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * ==========================================================================================
+ * =                            JAHIA'S ENTERPRISE DISTRIBUTION                             =
+ * ==========================================================================================
+ *
+ *                                  http://www.jahia.com
+ *
+ * JAHIA'S ENTERPRISE DISTRIBUTIONS LICENSING - IMPORTANT INFORMATION
+ * ==========================================================================================
+ *
+ *     Copyright (C) 2002-2019 Jahia Solutions Group. All rights reserved.
+ *
+ *     This file is part of a Jahia's Enterprise Distribution.
+ *
+ *     Jahia's Enterprise Distributions must be used in accordance with the terms
+ *     contained in the Jahia Solutions Group Terms &amp; Conditions as well as
+ *     the Jahia Sustainable Enterprise License (JSEL).
+ *
+ *     For questions regarding licensing, support, production usage...
+ *     please contact our team at sales@jahia.com or go to http://www.jahia.com/license.
+ *
+ * ==========================================================================================
+ */
+package org.jahia.modules.graphql.provider.dxm.util;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.jackrabbit.value.BinaryImpl;
+import org.jahia.api.Constants;
+import org.jahia.modules.graphql.provider.dxm.DataFetchingException;
+import org.jahia.services.content.JCRNodeIteratorWrapper;
+import org.jahia.services.content.JCRNodeWrapper;
+import org.jahia.services.content.JCRSessionWrapper;
+import org.jahia.services.content.decorator.JCRFileContent;
+
+import javax.jcr.RepositoryException;
+import java.io.*;
+import java.util.List;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+import java.util.zip.ZipOutputStream;
+
+/**
+ * helper methods to zip folders/files
+ *
+ * @author yousria
+ */
+public class ZipUtils {
+
+
+    /**
+     * method to zip a file / several files / folder or add files to an existing zip file (without duplicates)
+     *
+     */
+    public static void addToZip(List<JCRNodeWrapper> nodes, JCRNodeWrapper file) {
+        InputStream is = null;
+        String filename = file.getName();
+        try {
+            FileOutputStream fos = new FileOutputStream(filename);
```

Comment:
Do not create temporary files in current folder , if a temporary file is needed use File.createTempFile() and delete it as the end !